core/lib/server/pub.ts

```diff
@@ -1678,6 +1747,39 @@ export async function getPubsWithRelatedValues<Options extends GetPubsWithRelate
 										.over((ob) => ob.partitionBy("pv.fieldId"))} desc`,
 								"pv.rank",
 							])
+							.$if(Boolean(allowedPubTypes?.length), (qb) =>
+								qb
+									.leftJoin(
+										"pubs as related_pubs",
+										"related_pubs.id",
+										"pv.relatedPubId"
+									)
+									.where((eb) =>
+										eb.or([
+											eb("related_pubs.pubTypeId", "is", null),
+											eb("related_pubs.pubTypeId", "in", allowedPubTypes!),
+										])
+									)
+							)
+							.$if(Boolean(allowedStages?.length), (qb) =>
+								qb
+									.leftJoin(
+										"PubsInStages as related_pubs_in_stages",
+										"related_pubs_in_stages.pubId",
+										"pv.relatedPubId"
+									)
+									.where((eb) =>
+										eb.or([
+											// shouldnt check for stages for normal pub values
+											eb("pv.relatedPubId", "is", null),
+											stagesWhere(
+												eb,
+												allowedStages!,
+												"related_pubs_in_stages.stageId"
+											),
+										])
+									)
+							)
```

Comment:
we only filter the relatedPubs by the _allowed_ pubtypes/stages, not by the request pubtypes/stages. in the future we might want to do that, but i think it would be confusing for now, bc i expect this

"give me all submissions and their related values" -> `GET /pubs?pubTypeId=<submissionId>`

to give me all related pubs, even if theyre not submissions.

the allowedPubTypes should filter that, bc it's more of a "hiding" purpose than a "find" purpose.

but in the future we should make easier to do, its quite confusing

Reply:
edit: ive added this as a comment